AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -723,6 +724,21 @@ private static void handleApiResponse(JSONObject response, List<Map<String, Obje
       for (int i = requests.size() - 1; i >= 0; i--) {
         Map<String, Object> currentRequest = requests.get(i);
         if (Constants.Methods.START.equals(currentRequest.get(Constants.Params.ACTION))) {
+          if (currentRequest.containsKey(Constants.Params.REQ_ID)) {
+              for (int j = Request.numResponses(response) - 1; j >= 0; j--) {
+                JSONObject currentResponse = Request.getResponseAt(response, j);
+                try {
+                  if (currentResponse.get(Constants.Params.REQ_ID) ==
+                          currentRequest.get(Constants.Params.REQ_ID)) {
+                    lastStartResponse = currentResponse;
```

Comment:
Should we do the check on 742 first? It's just going to overwrite `lastStartResponse` when `i < responseCount`. I can't tell off-hand how often/likely that is. Also we could skip this loop.